docs/src/cli/deploy-a-program.md

```diff
@@ -282,14 +282,24 @@ intermediary buffer contents and then vote to allow an upgrade using it.
 solana program write-buffer <PROGRAM_FILEPATH>
 ```
 
-Buffer accounts support authorities like program accounts:
+Buffer accounts are managed by some authority - default signer, or signer specified with 
+`--buffer-authority` option:
+
+```bash
+solana program write-buffer <PROGRAM_FILEPATH> --buffer-authority <BUFFER_AUTHORITY_SIGNER>
+```
+
+Note, `--buffer-authority` from above must be a **Signer** - this prevents a developer from giving 
+upgrade authority to a key that they do not have access to. That means you can't set `--buffer-authority` 
+in `solana program write-buffer` to a Pubkey, and hence for certain use-cases (like offline signing) you 
+might want to use `solana program set-buffer-authority` command to set the authority for existing buffer 
+(which supports authority in the form of Pubkey, or even a program account):
```

Comment:
A few nits to be more forceful and terse, what do you think?

```suggestion
Only the buffer authority may write to the buffer, so the `--buffer-authority` above must be a
**signer**, and not an address.  This requirement limits usage with offline signers.

To use an offline address as a buffer authority, the buffer account must be initialized and
written with an online keypair, and then the buffer authority must be assigned using
`solana program set-buffer-authority`:
```

Reply:
Sounds good! Applied in https://github.com/solana-labs/solana/pull/34366/commits/d35fbb099feb14301f9fb42e5acf252fa2af0a5b